### PR TITLE
fix: Improve OLM installer to re-use in the CodeReady.

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this PR do?
Improve OLM installer to re-use in the CodeReady.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16862

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

File: src/api/kube.ts [hunk 1]
```diff
@@ -1151,12 +1151,14 @@ export class KubeHelper {
       }
       if (flags.tls) {
         yamlCr.spec.server.tlsSupport = flags.tls
-        if (!yamlCr.spec.k8s.tlsSecretName) {
+        if (yamlCr.spec.k8s && !yamlCr.spec.k8s.tlsSecretName) {
```
Comment (tolusha) [inline src/api/kube.ts:1154]:
```suggestion
        if (!yamlCr.spec.k8s) {
          yamlCr.spec.k8s = {}
        }
        if (!yamlCr.spec.k8s.tlsSecretName) {
```

File: src/api/kube.ts [hunk 2]
```diff
@@ -1151,12 +1151,14 @@ export class KubeHelper {
...
           yamlCr.spec.k8s.tlsSecretName = 'che-tls'
         }
       }
       yamlCr.spec.server.selfSignedCert = flags['self-signed-cert']
-      yamlCr.spec.k8s.ingressDomain = flags.domain
+      if (yamlCr.spec.k8s) {
```
Comment (tolusha) [inline src/api/kube.ts:1159]:
```suggestion
      if (flags.domain) {
```

File: src/api/kube.ts [hunk 3]
```diff
@@ -1150,13 +1150,18 @@ export class KubeHelper {
...
+          yamlCr.spec.k8s = {}
+        }
         yamlCr.spec.server.tlsSupport = flags.tls
         if (!yamlCr.spec.k8s.tlsSecretName) {
           yamlCr.spec.k8s.tlsSecretName = 'che-tls'
         }
       }
       yamlCr.spec.server.selfSignedCert = flags['self-signed-cert']
-      yamlCr.spec.k8s.ingressDomain = flags.domain
+      if (flags.domain) {
```
Comment (tolusha) [inline src/api/kube.ts:1162]:
```suggestion
        if (!yamlCr.spec.k8s) {
          yamlCr.spec.k8s = {}
        }
      if (flags.domain) {
```
Sorry, I forgot one more changes.

Comment (tolusha) [reply src/api/kube.ts:1165]:
```suggestion
```

**Comment by AndrienkoAleksandr** [comment]
[build]

**Comment by AndrienkoAleksandr** [comment]
[test]

**Comment by AndrienkoAleksandr** [comment]
Looks like unable to pull devfile registry image from quay.io . ImagePullBackOff ...

**Comment by tolusha** [comment]
[test]

**Comment by tolusha** [comment]
[test]

**Comment by AndrienkoAleksandr** [comment]
Closed on favor https://github.com/che-incubator/chectl/pull/724

**Pull request closed without merging**